### PR TITLE
Ria 3454 notifications when application online decided

### PR DESCRIPTION
**Pull request by mkayad**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RIA-3454

### Change description ###

Notification when an application has been decided

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

**Review by tarun-palisetty — DISMISSED** [review]
LGTM

**Review by tarun-palisetty — APPROVED** [review]

File: src/functionalTest/resources/scenarios/RIA-3453-make-an-application-notification-home-office-apc-after-listing.json [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      "eventId": "makeAnApplication",
+      "state": "prepareForHearing",
+      "caseData": {
+        "template": "minimal-appeal-submitted.json",
+        "replacements": {
+          "currentCaseStateVisibleToHomeOfficeAll":"prepareForHearing",
+          "currentCaseStateVisibleToLegalRepresentative": "prepareForHearing",
+          "listCaseHearingCentre": "taylorHouse",
+          "ariaListingReference": "PA/12345/2019",
+          "makeAnApplications": [
```
Comment (makowalski) [inline src/functionalTest/resources/scenarios/RIA-3453-make-an-application-notification-home-office-apc-after-listing.json:17]: please add usecase where there are two objects in the list

File: src/functionalTest/resources/scenarios/RIA-3453-make-an-application-notification-home-office-apc-after-listing.json [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            {
+              "id": "1",
+              "value": {
+                "type":"withdraw",
+                "details":"",
+                "evidence":[],
+                "applicant":"",
+                "date":"{$TODAY}",
+                "decision":"GRANTED",
```
Comment (makowalski) [inline src/functionalTest/resources/scenarios/RIA-3453-make-an-application-notification-home-office-apc-after-listing.json]: what happens if the decision is REFUSED?

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/MakeAnApplicationService.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+@Service
+public class MakeAnApplicationService {
+    public String getMakeAnApplicationTypeName(AsylumCase asylumCase) {
+
+        Optional<List<IdValue<MakeAnApplication>>> maybeMakeAnApplications = asylumCase.read(AsylumCaseDefinition.MAKE_AN_APPLICATIONS);
+
+        if (maybeMakeAnApplications.isPresent()) {
+            List<IdValue<MakeAnApplication>> makeAnApplications = maybeMakeAnApplications.orElse(Collections.emptyList());
+            if (makeAnApplications.size() > 0) {
+                return makeAnApplications.get(0).getValue().getType();
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/MakeAnApplicationService.java:26]: is it always first in the row, what happens when the decision is refused?

**Review by makowalski — CHANGES_REQUESTED** [review]
please add more usecases where there is more than one application and use case when the application is refused.
You may consider removing `MakeAnApplicationService` because it is created only for `Optional` class handling

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/MakeAnApplicationService.java (hunk 3 above)
Comment (mkayad) [reply src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/MakeAnApplicationService.java:26]: yes, the current application is always added the beginning of the list:
see this file for reference
https://github.com/hmcts/ia-case-api/pull/568/files#diff-650d0010f8d40cbed667257b0a9503fa70dd0ccfd08c52b72994e8777a2e97f6

File: src/functionalTest/resources/scenarios/RIA-3453-make-an-application-notification-home-office-apc-after-listing.json (hunk 2 above)
Comment (mkayad) [reply src/functionalTest/resources/scenarios/RIA-3453-make-an-application-notification-home-office-apc-after-listing.json]: this scenario is handled in 3454 functional tests where the decision is important, IN 3453, we don't use the decision

Comment (makowalski) [reply src/functionalTest/resources/scenarios/RIA-3453-make-an-application-notification-home-office-apc-after-listing.json]: hmm, why it is GRANTED if the decision is not done yet? same like other fields:
```
                "decisionReason":"No Reason given",
                "decisionDate":"{$TODAY}",
                "decisionMaker": "Judge"
```

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/MakeAnApplicationService.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                        }
+                    }
+                }
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    public boolean isApplicationListed(State state) {
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/MakeAnApplicationService.java:57]: this method is a Business logic which is well placed in separate service

**Review by makowalski — APPROVED** [review]
LGTM

**Review by Srihari-Nakkala — APPROVED** [review]
LGTM

**Review by makowalski — APPROVED** [review]
LGTM

**Pull request merged**